activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -410,26 +410,27 @@ def schema_up_to_date?(configuration, format = ActiveRecord.schema_format, file
         end
       end
 
-      def reconstruct_from_schema(db_config, format = ActiveRecord.schema_format, file = nil) # :nodoc:
-        file ||= schema_dump_path(db_config, format)
+      def reconstruct_from_schema(db_config, file = nil) # :nodoc:
+        file ||= schema_dump_path(db_config, db_config.format)
 
         check_schema_file(file) if file
 
         with_temporary_pool(db_config, clobber: true) do
-          if schema_up_to_date?(db_config, format, file)
+          if schema_up_to_date?(db_config, file)
             truncate_tables(db_config) unless ENV["SKIP_TEST_DATABASE_TRUNCATE"]
           else
             purge(db_config)
-            load_schema(db_config, format, file)
+            load_schema(db_config, db_config.format, file)
           end
         rescue ActiveRecord::NoDatabaseError
           create(db_config)
-          load_schema(db_config, format, file)
+          load_schema(db_config, db_config.format, file)
         end
       end
 
-      def dump_schema(db_config, format = ActiveRecord.schema_format) # :nodoc:
+      def dump_schema(db_config, format) # :nodoc:
```

Comment:
Subtle, but this will break for any callers that omit the second parameter (there are a couple callers within Rails that do this). Especially since you want to default format to `db_config.schema_format` it's probably best to allow callers to just pass in the db_config without the format being required.